### DISCUSSION TITLE
RL integration with RQ, RT

### DISCUSSION THREAD

**Discussion by Moshyfawn** [TanStack/router#122] [Q&A] [answered]
I'm playing around with the wonderful Tanstack tools and their integration with each other.

It seems like each tool provides some sort of state management; RT has `usePagination`, `useSort` & `useFilters`, RQ can handle pagination with `useInfiniteQuery` and RL can obviously help store state in the URL. In the situation where some paginated data, that's retrieved using RQ that's displayed with the help of RT, what tool should take the responsibility to manage data params like pagination, sorting & filtering? I've come to the conclusion, that it's RL that's gonna operate the values and change search params accordingly. But I'm curious to see if there's any other way that'd prove to be better and in what other ways Tanstack tools can be integrated with each other, knowing that, say, Nozzle work with the stack and they sure have some established way of dong things.

There're a bunch of decisions to make like, should the button that "loads more data" change the URL so that RQ key change triggers a new fetch or call the `fetchNextPage` with new search params from RQ and update the URL search params at the same time? There're other things, that I can't think of rn

**Comment by tannerlinsley** [comment, marked as answer]
Generally, as much as makes sense should be stored in the URL that drives the state of table and query variables. This is pretty easy to do with React Query because it's API is 100% controlled for the most part. React Table v7 however is finicky still. Some of its state is internal and automanage, so it's hard to 100% control it via props. This is going to change in v8 (the Typescript rewrite I'm doing).
